featuretools/feature_discovery/LiteFeature.py

```diff
@@ -0,0 +1,295 @@
+
+    def __lt__(self, other: LiteFeature):
+        return self._id < other._id
+
+    def __ne__(self, other):
+        return self._id != other._id
+
+    def __hash__(self):
+        return hash(self._id)
+
+    def _generate_hash(self) -> str:
+        return self.hash(
+            name=self._name,
+            primitive=self._primitive,
+            base_features=self._base_features,
+            df_id=self._df_id,
+            idx=self._idx,
+        )
+
+    def get_primitive_name(self) -> Union[str, None]:
+        return self._primitive.name if self._primitive else None
+
+    def get_dependencies(self, deep=False) -> List[LiteFeature]:
+        flattened_dependencies = []
+        for f in self._base_features:
+            flattened_dependencies.append(f)
+
+            if deep:
+                dependencies = f.get_dependencies()
+                if isinstance(dependencies, list):
+                    flattened_dependencies.extend(dependencies)
+                else:
+                    flattened_dependencies.append(dependencies)
+        return flattened_dependencies
+
+    def get_origin_features(self) -> List[LiteFeature]:
+        all_dependencies = self.get_dependencies(deep=True)
+        return [f for f in all_dependencies if f._depth == 0]
+
+    @property
+    def column_schema(self) -> ColumnSchema:
+        return ColumnSchema(logical_type=self.logical_type, semantic_tags=self.tags)
+
+    def dependent_primitives(self) -> Set[Type[PrimitiveBase]]:
+        dependent_features = self.get_dependencies(deep=True)
+        dependent_primitives = {
+            type(f._primitive) for f in dependent_features if f._primitive
+        }
+        if self._primitive:
+            dependent_primitives.add(type(self._primitive))
+        return dependent_primitives
+
+    def to_dict(self) -> Dict[str, Any]:
+        return {
+            "name": self.name,
+            "logical_type": self.logical_type.__name__ if self.logical_type else None,
+            "tags": list(self.tags),
+            "primitive": hash_primitive(self.primitive)[0] if self.primitive else None,
+            "base_features": [x.id for x in self.base_features],
+            "df_id": self.df_id,
+            "id": self.id,
+            "related_features": [x.id for x in self.related_features],
+            "idx": self.idx,
+        }
+
+    def is_multioutput(self) -> bool:
+        return len(self._related_features) > 0
+
+    def copy(self) -> LiteFeature:
+        copied_feature = LiteFeature(
+            name=self._name,
+            logical_type=self._logical_type,
+            tags=self._tags,
+            primitive=self._primitive,
+            base_features=[x.copy() for x in self._base_features],
+            df_id=self._df_id,
+            idx=self._idx,
+            related_features=self._related_features,
+        )
+
+        copied_feature.name = self._alias
+
+        return copied_feature
+
+    def __repr__(self) -> str:
```

Comment:
Thoughts on simplifying the repr to be more similar to BaseFeatures? I think including all this info is making it hard to understand a short list of origin features (compare to the list of FeatureBase objects below)

<img width="1427" alt="image" src="https://user-images.githubusercontent.com/64278226/234041580-6ec92ded-4540-40fe-9e5b-1a6de2b30fae.png">